### PR TITLE
[JENKINS-23378] update to Jetty9.2

### PR DESCRIPTION
**Pull request by kohsuke**
`hpi:run` should run Jetty 9.2 to support Servlet 3.1.

I've fixed how we track upstream files from jetty-maven-plugin so that future merges like this will be more manageable.

The diff view in this PR mixes upstream changes in Jetty and local changes, which is useless. If you want to really understand the diff, look at `git log -U jetty9.2 '^incoming' '^master'`

@reviewbybees

### CODE REVIEW

**Comment by kohsuke** [comment]
This version runs git-plugin OK, so it's not dead on arrival.

File: src/main/java/org/jenkinsci/maven/plugins/hpi/RunMojo.java [hunk 1]
```diff
@@ -612,19 +651,13 @@ public void addJars(org.eclipse.jetty.util.resource.Resource lib) {
...
-
-                return con;
-            }
-        };
+    @Override
+    public void startJetty() throws MojoExecutionException {
+        if (httpConnector==null && defaultPort!=0) {
+            httpConnector = new MavenServerConnector();
+            httpConnector.setPort(defaultPort);
+        }
```
Comment (tfennelly) [inline src/main/java/org/jenkinsci/maven/plugins/hpi/RunMojo.java]: Maybe you meant this to be something like...

if (httpConnector==null) {
    httpConnector = new MavenServerConnector();
}
if (defaultPort!=0) {
    httpConnector.setPort(defaultPort);
}

**Comment by tfennelly** [comment]
:bee: from what I can see.

File: src/main/java/org/jenkinsci/maven/plugins/hpi/RunMojo.java (hunk 1 above)
Comment (kohsuke) [inline src/main/java/org/jenkinsci/maven/plugins/hpi/RunMojo.java]: This is as intended. As the javadoc of `defaultPort` states, if a connector is configured that'll take precedence.

**Pull request merged**